csrc/evaluator_common.cpp

```diff
@@ -170,9 +170,8 @@ void PrecomputedValues::bindInputs(const KernelArgumentHolder& args) {
       if (!tensor.is_cpu()) {
         bindTensorMetaData(tensor_input, tensor);
       }
-    } else {
-      bindValue(input->evaluatorIndex(), *args[i]);
     }
+    bindValue(input->evaluatorIndex(), *args[i]);
```

Comment:
Oops, I think I remembered why I felt worried about this. At some point, I tried to do the same thing, but eventually gave up. Because doing so will make `PrecomputedValues` hold ownership of tensors, and as a result, memory usage will go up a lot.

Reply:
Ah that makes sense. So we don't want `PrecomputedValues` or `ExpressionEvaluator` to own any tensor buffers, but it might be nice to still hold them. We could bind a `Pointer` instead but I think we might still wind up owning the tensor in the `PolymorphicValue` that points to. We could also introduce a `TensorMeta` class that doesn't own its data but holds sizes, strides, dtype, device, and data pointer. I am not sure whether that would slow down evaluation of CPU aten ops though, since we'd need to construct an `at::Tensor` when evaluating those ops then.

Reply:
OK I'm trying to understand the existing `TensorMetaData` class now. I think maybe we could bind these values as the the outputs of `IrBuilder::metadataExpr(tv)` instead of binding to `tv` directly?

Reply:
I think it is OK for `ExpressionEvaluator` to bind tensors, actually, we are evaluating tensor expressions. Because we do not carry instances of `ExpressionEvaluator` for long time.

Reply:
But if we bind `tens` to `tv` in `ExpressionEvaluator ee` then later we bind a `PrecomputedValues pv` to `ee`, then we `evaluate` an expression with `tv` as an input, won't that populate `pv` with `tens`?

Reply:
No, I don't think so.  `PrecomputedValues` -> `ExpressionEvaluator` is one-directional.

https://github.com/NVIDIA/Fuser/blob/5ff26b9ed9cc6d59b92490ba535c366e91747f28/csrc/expr_evaluator.cpp#L162-L166

Reply:
Got it. I'll revert the last PR then and add a note that we bind to `ExpressionEvaluator` but should avoid binding to `PrecomputedValues`.